core/domain/stats_domain.py

```diff
@@ -465,6 +578,8 @@ def __eq__(self, other):
         Returns:
             bool. Whether the two instances have the same values.
         """
+        if not isinstance(other, StateStats):
+            return NotImplemented
```

Comment:
Where is NotImplemented defined?

Also, why do we add this rather than declaring "other" to have StateStats type?

Reply:
Done (deleted 581-582)